src/hooks/keyboardContext.tsx

```diff
@@ -0,0 +1,37 @@
+
+import React, { createContext, useState } from 'react';
+import AudioContextWrapper from '../classes/audioContextWrapper';
+import waveforms from "../data/waveforms";
+
+type UpdateKeyboardContextType = {
+    updateKeyboardState: () => void | undefined;
+  };
+
+const defaultOctave = 4;
+const defaultState = {
+    octave: defaultOctave,
+    setIsClicked: false,
+    wave: waveforms.sine,
+    audioContextWrapper: new AudioContextWrapper()
+  }
+
+type UpdateKeyboardValues = {
+    updateKeyboardState: () => void;
+}
+
+const KeyboardContext = React.createContext<any>({...defaultState});
+const UpdateKeyboardContext = createContext<any>(undefined);
+
+export function KeyboardProvider({children} : any){
+const [keyboardState, updateKeyboardState] = useState({...defaultState});
+
+    return (
+        <UpdateKeyboardContext.Provider value = {updateKeyboardState}>
+            <KeyboardContext.Provider value={ keyboardState }>
+                {children}
+            </KeyboardContext.Provider>
+        // </UpdateKeyboardContext.Provider>
```

Comment:
Why isn't this needed?

Reply:
Thank you for finding this - it was driving me crazy. 
We did need it. Fortunately, // doesn't add comments in the code, so instead it was rendering "//" to the screen